src/server/svr_jobfunc.c

```diff
@@ -3029,11 +3029,31 @@ Time4occurrenceFinish(resc_resv *presv)
 	/* If the start time of the reservation was altered, copy from RESV_ATR_start
  	 * will make the next instance to have it's start time altered so take the start
  	 * time from the ri_alter_stime. */
-	if (presv->ri_alter_stime) {
-		dtstart = presv->ri_alter_stime;
-		presv->ri_alter_stime = 0;
+	if (presv->ri_alter.ra_revert.rr_stime) {
+		dtstart = presv->ri_alter.ra_revert.rr_stime;
+		presv->ri_alter.ra_revert.rr_stime = 0;
```

Comment:
I don't understand this part. I assume that when this function is called, the reservation is in BEING_ALTERED state and that is why we want to register the next reservation task with the original stime and select rather than using unconfirmed parameters.
I'm confused why are we resetting the original stime and select to NULL? won't we need them in case the confirmation fails? Also, if resetting the value is necessary, don't we also have to reset the flags?

Reply:
This is for a standing reservation occurrence that has been modified and is currently ending.  The state will be running.  We need to revert the start time, duration, and select back to the original values.  Those values are then used to move to the next occurrence.  We set the values to NULL because once reset back to the original values, we don't need them any more.  When we move onto the 3rd occurrence, they will already be the correct values.